cmd/guacingest/cmd/ingest.go

```diff
@@ -179,9 +177,26 @@ func getIngestor(ctx context.Context, transportFunc func([]assembler.IngestPredi
 	}, nil
 }
 
-func getAssembler(ctx context.Context, opts options) (func([]assembler.IngestPredicates) error, error) {
-	httpClient := http.Client{}
-	gqlclient := graphql.NewClient(opts.graphqlEndpoint, &httpClient)
-	f := helpers.GetAssembler(ctx, gqlclient)
-	return f, nil
+func getAssembler(ctx context.Context, graphqlEndpoint string) func([]assembler.IngestPredicates) error {
+	// Same as https://pkg.go.dev/net/http#DefaultTransport but with greater
+	// MaxIdleConnsPerHost so that we effectively re-use connections when doing
+	// parallel ingestion calls.
+	var dialer = &net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}
+	var customTransport http.RoundTripper = &http.Transport{
```

Comment:
nit: feel like these would be nice if they were changable via config file, would help in perf testing as well.

Reply:
Sure can do that. The only modification that was made here was with `MaxIdleConnsPerHost`. Did we want to expose the others also? cc @jeffmendoza

Reply:
Just a "parallel: yes/no" should be fine imo.

But if you want to add the # outstanding config, it would be here: https://github.com/guacsec/guac/blob/main/pkg/assembler/clients/helpers/parallel.go#L37-L39 Then the `MaxIdleConnsPerHost` needs to be greater than that by some margin.